src/Jackalope/Node.php

```diff
@@ -1564,12 +1575,24 @@ public function getIterator()
      * @see Node::refresh
      * @see Node::__construct
      */
-    protected function _setProperty($name, $value, $type, $internal)
+    protected function loadProperty($name)
     {
+        if (isset($this->properties[$name])) {
+            return $this->properties[$name];
+        }
+
+        if (!isset($this->propertyValues[$name])) {
+            return;
+        }
+
         if ($name == '' | false !== strpos($name, '/')) {
             throw new InvalidArgumentException("The name '$name' is no valid property name");
         }
 
+        $value = $this->propertyValues[$name]['value'];
+        $type = $this->propertyValues[$name]['type'];
+        $internal = $this->propertyValues[$name]['internal'];
+
         if (!isset($this->properties[$name])) {
```

Comment:
I know, but I am sure that the code in the else path was there for a reason. Maybe there's a caveat in this new strategy? But if you say it should be save to delete I'll do it.